src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -0,0 +1,299 @@
+
+  actor_table_data->set_state(rpc::ActorTableData::PENDING);
+  actor_table_data->mutable_task_spec()->CopyFrom(request.task_spec());
+
+  actor_table_data->mutable_address()->set_raylet_id(ClientID::Nil().Binary());
+  actor_table_data->mutable_address()->set_worker_id(WorkerID::Nil().Binary());
+
+  auto actor = std::make_shared<GcsActor>(*actor_table_data);
+  // The backend storage is reliable in the future, so the status must be ok.
+  RAY_CHECK_OK(actor_info_accessor_.AsyncUpdate(
+      actor_id, actor_table_data, [this, actor](Status status) {
+        RAY_CHECK_OK(status);
+        registered_actors_.emplace(actor->GetActorID(), actor);
+        gcs_actor_scheduler_->Schedule(actor);
+        // Invoke all callbacks for all registration requests of this actor (duplicated
+        // requests are included) and remove all of them from
+        // actor_to_register_callbacks_.
+        auto iter = actor_to_register_callbacks_.find(actor->GetActorID());
+        RAY_CHECK(iter != actor_to_register_callbacks_.end() && !iter->second.empty());
+        for (auto &callback : iter->second) {
+          callback(actor);
+        }
+        actor_to_register_callbacks_.erase(iter);
+      }));
+}
+
+void GcsActorManager::ReconstructActorOnWorker(const ray::ClientID &node_id,
+                                               const ray::WorkerID &worker_id,
+                                               bool need_reschedule) {
+  std::shared_ptr<GcsActor> actor;
+  // Cancel the scheduling of the related actor.
+  auto actor_id = gcs_actor_scheduler_->CancelOnWorker(node_id, worker_id);
+  if (!actor_id.IsNil()) {
+    auto iter = registered_actors_.find(actor_id);
+    if (iter != registered_actors_.end()) {
+      actor = iter->second;
+    }
+  } else {
+    // Find from worker_to_created_actor_.
+    auto iter = worker_to_created_actor_.find(worker_id);
+    if (iter != worker_to_created_actor_.end()) {
+      actor = std::move(iter->second);
+      // Remove the created actor from worker_to_created_actor_.
+      worker_to_created_actor_.erase(iter);
+      // remove the created actor from node_to_created_actors_.
+      auto node_iter = node_to_created_actors_.find(node_id);
+      RAY_CHECK(node_iter != node_to_created_actors_.end());
+      node_iter->second.erase(actor->GetActorID());
+      if (node_iter->second.empty()) {
+        node_to_created_actors_.erase(node_iter);
+      }
+    }
+  }
+  if (actor != nullptr) {
+    // Reconstruct the actor.
+    ReconstructActor(actor, need_reschedule);
+  }
+}
+
+void GcsActorManager::ReconstructActorsOnNode(const ClientID &node_id) {
+  // Cancel the scheduling of all related actors.
+  auto scheduling_actor_ids = gcs_actor_scheduler_->CancelOnNode(node_id);
+  for (auto &actor_id : scheduling_actor_ids) {
+    auto iter = registered_actors_.find(actor_id);
+    if (iter != registered_actors_.end()) {
+      // Reconstruct the canceled actor.
+      ReconstructActor(iter->second);
```

Comment:
```suggestion
      ReconstructActor(std::move(iter->second));
```

Reply:
> 

std:: move cannot be used here, which will cause the actor in register_actors to be set to nullptr.